PA_real_time_monitoring.md

```diff
@@ -77,53 +77,112 @@ const BIDDING_SLOW_WEIGHT = 0.1;
      ...
   } catch (error) {
-    realTimeReporting.contributeToRealTimeHistogram(
-      BIDDING_THREW_BUCKET,
-      { priorityWeight: BIDDING_THREW_WEIGHT });
+    realTimeReporting.contributeToHistogram(
+      { bucket: BIDDING_THREW_BUCKET,
+        priorityWeight: BIDDING_THREW_WEIGHT });
     return;
   }
   
-  if (bid.bid > MAX_REASONABLE_BID) {
-    realTimeReporting.contributeToRealTimeHistogram(
+  if (bidOut.bid > MAX_REASONABLE_BID) {
+    realTimeReporting.contributeToHistogram(
       BID_TOO_HIGH_BUCKET,
-      { priorityWeight: BID_TOO_HIGH_WEIGHT });
+      { bucket: BID_TOO_HIGH_BUCKET,
+        priorityWeight: BID_TOO_HIGH_WEIGHT });
     return;
   }
 
-  return bid;
+  return bidOut;
 }
 ```
 
 
 ## Platform contributions: reporting errors not detectable in worklet JS
 
-There may be some errors that occur that are not visible in either `scoreAd()` or `generateBid()`. These include things like failures to fetch the bidding script, trusted real-time signals, or creative URL. In these cases, we propose to allocate a certain portion of the resulting histogram for platform errors. While this can be done as a completely separate set of buckets that are immutable from JavaScript, we would still require the number of total contributions across all buckets to be capped at 1, so these platform-assisted contributions will come with a default weighting and participate in the prioritization algorithm just like regular contributions.
+There may be some errors that occur that are not visible in either `scoreAd()` or `generateBid()`. These include things like failures to fetch the bidding script, trusted real-time signals, or creative URL. In these cases, we allocate a certain portion of the resulting histogram for platform errors. While this can be done as a completely separate set of buckets that are immutable from JavaScript, we would still require the number of total contributions across all buckets to be capped at 1, so these platform-assisted contributions will come with a default weighting and participate in the prioritization algorithm just like regular contributions.
 
-The priority weight of platform contributions will be hardcoded and specified, so you can reason about the behavior across platform and developer contributions.
+The priority weight of platform contributions is hardcoded as 1, so you can reason about the behavior across platform and developer contributions.
 
+Platform contributions are reported in the report’s `platformHistogram` field, separately from regular contributions. See [below](#sending-reports-after-an-auction) for more details.
+
+| Bucket | Error |
+| --- | --- |
+| 0 | Bidding script fetch error |
+| 1 | Scoring script fetch error |
+| 2 | Trusted bidding signals fetch error |
+| 3 | Trusted scoring signals fetch error |
+
+Fetch errors include non-2xx response code, response header does not have “Ad-Auction-Allowed: true”, response body cannot be parsed as valid json for trusted signals, etc,.
```

Comment:
A 304 (Not Modified) would be considered an error?

Reply:
yes. Non-2xx codes will be considered as errors.
See step 2 of spec: https://wicg.github.io/turtledove/#validate-fetching-response-headers

Reply:
Bidding and scoring scripts are expected to be slowly changing and adtechs will count on browsers' HTTP cache.

Reply:
[Here](https://crsrc.org/c/content/services/auction_worklet/public/cpp/auction_downloader.cc;drc=b8ddddd3dfef89a597a4e89d8cc76bcf8a15d655;l=403) is the code that checks for the header to be 2xx.
@morlovich knows the network related code better. @morlovich would you mind helping here?

Reply:
I think you normally won't see a 304 at this layer, but rather the cached response (with updated headers).

Probably this stuff:
https://fetch.spec.whatwg.org/#:~:text=If%20the%20revalidatingFlag%20is%20set%20and%20forwardResponse%E2%80%99s%20status%20is%20304%2C%20then%3A